unstructured/partition/pdf.py

```diff
@@ -243,6 +247,8 @@ def _partition_pdf_or_image_local(
     extract_element_types: Optional[List[str]] = None,
     image_output_dir_path: Optional[str] = None,
     pdf_image_dpi: Optional[int] = None,
+    analysis: bool = False,
+    analyzed_image_output_dir_path: Optional[str] = None,
     **kwargs,
 ) -> List[Element]:
     """Partition using package installed locally"""
```

Comment:
This function is excessively long and complex. It's got to be impossible to test thoroughly and take a long time to understand its behavior.

Maybe now isn't the right moment, but this is ripe for refactoring into a composition of much smaller components.